core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -335,31 +337,31 @@ private[spark] class TaskSchedulerImpl(
     var reason: Option[ExecutorLossReason] = None
     synchronized {
       try {
-        if (state == TaskState.LOST && taskIdToExecutorId.contains(tid)) {
```

Comment:
To understand this change, note that `taskIdToExecutorId` and `taskIdToTaskSetManager` should always have the same key set, so the `case Some(taskSet) if state == TaskState.LOST` is equivalent to this `if` statement.